R/bplapply-methods.R

```diff
@@ -45,58 +45,39 @@ setMethod("bplapply", c("ANY", "list"),
     if (!length(X))
         return(.rename(list(), X))
 
-    if (!bpschedule(BPPARAM) || length(X) == 1L || bpnworkers(BPPARAM) == 1L) {
-        param <- as(BPPARAM, "SerialParam")
-        return(bplapply(X, FUN, ..., BPREDO=BPREDO, BPPARAM=param))
-    }
-
     ## which need to be redone?
     redo_index <- .redo_index(X, BPREDO)
     if (any(redo_index)) {
-        X <- X[redo_index]
-        compute_element <- redo_index
+        compute_element <- which(redo_index)
```

Comment:
it is kind of stale as only the first version of `bplapply` requires it, but I think it could provide some performance benefit(but might be trivial) since subset by index should be faster than subset by the logical vector, so I just leave it here.